aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -322,6 +322,17 @@ void bernoulli_mkl_kernel(Tensor &self, const double p, Generator* gen) {
 }
 #endif
 
+static void exponential_kernel(TensorIterator& iter, double lambda, Generator* gen) {
+  AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "exponential_cpu", [&]() {
+    CPUGenerator* generator = get_generator_or_default<CPUGenerator>(gen, detail::getDefaultCPUGenerator());
+    std::lock_guard<std::mutex> lock(generator->mutex_);
+    cpu_serial_kernel(iter, [lambda, generator]() -> scalar_t {
+      at::exponential_distribution<double> exponential(lambda);
```

Comment:
`at::exponential_distribution<double> exponential(lambda);` should be outside of loop statement